src/analyses/dependence_graph.cpp

```diff
@@ -67,42 +53,54 @@ void dep_graph_domaint::control_dependencies(
 {
   // Better Slicing of Programs with Jumps and Switches
   // Kumar and Horwitz, FASE'02:
-  // Node N is control dependent on node M iff N postdominates one
-  // but not all of M's CFG successors.
+  // "Node N is control dependent on node M iff N postdominates, in
+  // the CFG, one but not all of M's CFG successors."
   //
-  // candidates for M are from and all existing control-depended on
-  // nodes; from is added if it is a goto or assume instruction
-  if(from->is_goto() ||
-     from->is_assume())
-    control_deps.insert(from);
+  // The "successor" above refers to an immediate successor of M.
+
+  // Candidates for M for "to" are "from" and all existing control
+  // dependencies on nodes. "from" is added if it is a goto or assume
```

Comment:
"all existing control dependencies on nodes" is less clear than the original phrasing I think. How about "all nodes X such that N is control-dependent on X"

Reply:
I just realized that this comment is not accurate anymore in the fixed version. Now we consider all control statements in the current function from which there is a path to the current statement (i.e., `to`) as potential sources of control dependency edges. I'll add a comment to explain this.